dask/dataframe/core.py

```diff
@@ -5138,17 +5138,20 @@ def cov_corr_chunk(df, corr=False):
         with warnings.catch_warnings(record=True):
             warnings.simplefilter("always")
             mu = (sums / counts).T
-        m = np.zeros(shape)
+        m = np.zeros_like(df.values, shape=shape)
         mask = df.isnull().values
         for idx, x in enumerate(df):
-            # Use .values to get the ndarray for the ufunc.
-            mu_discrepancy = np.subtract.outer(df.iloc[:, idx].values, mu[idx]) ** 2
+            # Avoid using ufunc.outer, since it is not supported by cupy
+            mu_discrepancy = np.zeros_like(df.values, shape=(len(df), len(mu)))
+            for j in range(len(mu)):
+                mu_discrepancy[:, j] = np.subtract(df.iloc[:, idx].values, mu[idx, j])
+            mu_discrepancy = mu_discrepancy ** 2
```

Comment:
I wonder if you might be able to do this instead with slicing with new dimensions, something like this ...

```python
x[:, None] - y[None, :]
```

Reply:
Yes- good idea